src/telemetry/telemetry_buffer.ts

```diff
@@ -0,0 +1,79 @@
+// Interfaces
+interface TelemetryEvent  {
+    timestamp : number,
+    action: TypeGenieTelemetryBuffer.Action,
+    value: any
+}
+
+interface TypeGenieTelemetryBufferInterface {
+
+    createTelemetryEvent(timestamp: number, action: TypeGenieTelemetryBuffer.Action, value: any) : TelemetryEvent;
+    addEvent(event: TelemetryEvent) : void;
+    removeEvent(index: number) : void;
+
+}
+
+
+
+export class TypeGenieTelemetryBuffer implements TypeGenieTelemetryBufferInterface{
+
+    private _events : TelemetryEvent[];
```

Comment:
I would suggest that the events be tied to a particular `session_id`. That way, we can associate the events to a particular `session_id`. The `session_id` in-turn contains all the relevant client information like which account_id, deployment_id, and user_id and at what time the session is created. 

P.S. - I don't think we save the `session_id` at the moment in the mongoDB database. So implementation would mean that when the session is created, we log that in the database along with metadata such as account_id, deployment_id, and user_id.

Reply:
Makes sense! Will take this into consideration in further updates